dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -46,6 +47,13 @@ def __str__(self):
     __repr__ = __str__
 
     def gpu(self):
+        if _conditionals.conditionals_enabled():
+            # Treat it the same way as regular operator would behave
+            [self_split], _ = _conditionals.apply_conditional_split_to_args([self], {})
+            transferred_node = DataNodeDebug(self._data._as_gpu(), self_split.name, "gpu",
+                                             self_split.source)
+            _conditionals.register_data_nodes(transferred_node, [self])
```

Comment:
Nitpick: is it really always transferred? If the node is already on GPU, it's a no-op, so perhaps calling it a `gpu_node` would be more accurate.

Reply:
I see a typo as well, I should copy the split node, adjusted + a shortcut.